File: sdk-python/tests/iwfcompat/state_failure_flow.py
# Portions of this file are derived from indeedeng/iwf-java-sdk.
# Those portions are licensed under the Apache License, Version 2.0.
# See LICENSES/Apache-2.0.txt and LEGACY_NOTICES.md.
#
# Modifications Copyright (c) 2026 Super Durable, Inc.
#
# Modifications are licensed under the Super Durable Source License 1.0.
# Third-Party Materials remain under the Apache License, Version 2.0.
# See LICENSE and LEGACY_NOTICES.md.

from dex import Context, Flow, Step, StepDecision, StepList, Wait


class StateFailureStep(Step[int]):
    def wait_for(self, context: Context, input: int) -> Wait:
        del context, input
        return Wait.skip_immediately()

    def execute(self, context: Context, input: int) -> StepDecision:
        del context, input
        raise RuntimeError("state API failure")


class StateFailureFlow(Flow[int]):
    start = StateFailureStep()

    def get_steps(self) -> StepList[int]:
        return StepList.start_step(self.start)

File: sdk-typescript/test/iwfcompat/state_failure_flow.ts
// Copyright (c) 2026 Super Durable, Inc.
//
// Licensed under the Super Durable Source License 1.0.
// You may not use this file except in compliance with the License.
// See the LICENSE file in the repository root.
//
// SPDX-License-Identifier: LicenseRef-Super-Durable-1.0

import {
  StepDef,
  Wait,
  doubleCodec,
  type Context,
  type Flow,
  type Step,
  type StepDecision,
} from "../../src/index.js";

class StateFailureStep implements Step<number> {
  public readonly inputCodec = doubleCodec;

  public getStepType(): string {
    return "StateFailureStep";
  }

  public waitFor(_context: Context, _input: number): Wait {
    return Wait.skipImmediately();
  }

  public execute(_context: Context, _input: number): StepDecision {
    throw new Error("state API failure");
  }
}

export class StateFailureFlow implements Flow<number> {
  private readonly start = new StateFailureStep();

  public getFlowType(): string {
    return "StateFailureFlow";
  }

  public getSteps() {
    return [StepDef.startStep(this.start)];
  }
}
